### ISSUE TITLE
Date field with "date_format" validation does not work correctly on edit.

### ISSUE THREAD

**Issue by the-artisan** [laravel/nova-issues#3284]
- Laravel Version: 7.28.4
- Nova Version: 3.3.2
- PHP Version: 7.3.21

### Description:

Date field with "date_format" rule does not validate while editing a content with field data already set. You need at least to click on the input field to trigger a "change" event. It looks like the default value of Date fields when visiting the edit page of a content is set as Y-m-d.

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create a resource with a "Date" field having this formatting and rules:

```
Date::make('Start date', 'start_date')
                ->format('DD-MM-YYYY')
                ->pickerFormat('d-m-Y')
                ->rules('required','date_format:d-m-Y'),
```

2. Add a new content of that resource then edit it.
3. Without focusing on the date field update the resource. You should receive a validation error on the date field.

If we use this validation rules:

```
->rules('required','date_format:Y-m-d')
```

saving without any edit to the date field will work.

**Comment by crynobone** [comment]
`Date` and `DateTime` will always cast to `Y-m-d` and `Y-m-d H:i:s.u` from server-side and `pickerFormat()` feature is misleading and has no effect at the moment.

I would strongly suggest just changing the display format using `->pickerDisplayFormat('d-m-Y')` instead if you want the date to display that way. 

And I will suggest `pickerFormat()` to be remove for next series.

**Comment by the-artisan** [comment]
It looks like pickerDisplayFormat is not available on the 3.3.2 version. I guess i will stick with the Y-m-d format before updating to a more recent version.

**Comment by monaye** [comment]
@the-artisan it's a workaround but you can make both validation and mutator to accept both formats to get it working.
